playbooks/create-job.yml

```diff
@@ -1,58 +1,88 @@
-- hosts: localhost
+- hosts: all
   become: true
-
-
   vars:
     ansible_python_interpreter: /usr/bin/python3.10
-    jenkins_host: 127.0.0.1
-    jenkins_port: 8085
-    jenkins_admin_user: admin
-    jenkins_admin_pass: admin
+    jenkins_install_lts: true
+    jenkins_http_port: 8085
+    jenkins_admin_username: admin
+    jenkins_admin_password: admin
     java_packages:
       - openjdk-17-jdk
     jenkins_maven_installations:
       - 3.8.4
       - 3.9.0
-    jenkins_job_config: "{{ lookup('template', 'job-config-2.xml.j2') }}"
     jenkins_maven_template: "{{ lookup('template', 'maven-install.groovy.j2') }}"
-    jenkins_job_name: spipeline0
+    jenkins_job_config: "{{ lookup('template', 'job-config-2.xml.j2') }}"
+    jenkins_multibranch_job_config: "{{ lookup('template', 'multibranch-pipeline-config.xml.j2') }}"
+    jenkins_job_name: pipeline-job
+    jenkins_multibranch_job_name: multi-pipeline-job
 
   pre_tasks:
     - include_vars: plugins.yml
+      tags: always
+    - name: Set jenkins_repo_key_latest_url
+      ansible.builtin.set_fact:
+        jenkins_repo_key_latest_url: "https://pkg.jenkins.io/{{ ansible_os_family | lower }}{{ '-stable' if (jenkins_install_lts | bool) else '' }}/jenkins.io-2023.key"
+      when: ansible_os_family == 'Debian' or ansible_os_family == 'RedHat'
+      tags: always
 
   roles:
-    - {role: geerlingguy.jenkins, jenkins_hostname: "{{ jenkins_host }}", jenkins_http_port: "{{ jenkins_port }}", jenkins_admin_username: "{{ jenkins_admin_user }}", jenkins_admin_password: "{{ jenkins_admin_pass }}", jenkins_plugins: "{{ plugins }}"}
-    - {role: geerlingguy.java, become: yes, java_packages: openjdk-17-jdk}
-  tags: notest
+    - { role: geerlingguy.git, tags: always }
+    - { role: geerlingguy.java, tags: always }
+    - { role: geerlingguy.jenkins, tags: [ never, step1 ], jenkins_prefer_lts: "{{ jenkins_install_lts }}", jenkins_repo_key_url: "{{ jenkins_repo_key_latest_url }}" }
+    - { role: geerlingguy.jenkins, tags: [ step2, step3, step4, step5, step6 ], jenkins_plugins: "{{ plugins }}", jenkins_prefer_lts: "{{ jenkins_install_lts }}", jenkins_repo_key_url: "{{ jenkins_repo_key_latest_url }}" }
 
   post_tasks:
+    - name: Install Python pip
+      apt: name={{ item }} update_cache=true state=present force_apt_get=yes
```

Comment:
Please do not use apt, use package instead (we need to adapt our playbook also to CentOs and Rocky)